### ISSUE TITLE
Confirm PM senders

### ISSUE THREAD

**Issue by PeterSurda** [Bitmessage/PyBitmessage#1482]
Similarly to how mobile chat programs work, there should be a way (maybe using the blacklist / whitelist, we could add greylist or something to the option) to confirm new contacts.

1. sender sends a message to a new recipient
2. new recipient sees the message, but he also has a popup for whether to allow or block the sender
3. after allowing, sender can be added to addressbook
4. after blocking, it could go to spam or something

The details can be changed later. It probably needs one backend implementation and multiple UI ones (API, QT, kivy, ...).

**Comment by g1itch** [comment]
Mobile chat programs require an authorization to show contact's status. Bitmessage doesn't send statuses.

Everything else you described here looks like just a special way to handle blacklist for personal messages. The new senders may be added into both black and white lists, so that any their messages except the first will go to "spam" folder until receiver deletes them from blacklist.

Such logic can be implemented in `class_ObjectProcessor` with minimum UI changes. In bitmessageqt it may be implemented as additional control (maybe just a button) added to `MessageList_AddressWidget`.

**Comment by PeterSurda** [comment]
> Everything else you described here looks like just a special way to handle blacklist for personal messages.

Yes, correct. It's mainly a usability issue, the backend changes wouldn't be very difficult.
